src/bootstrap/src/core/build_steps/tool.rs

```diff
@@ -1190,7 +1193,7 @@ fn run_tool_build_step(
 
         for add_bin in add_bins_to_sysroot {
             let bin_destination = bindir.join(exe(add_bin, target_compiler.host));
-            builder.copy_link(&tool_path, &bin_destination);
+            builder.copy_link_binary(&tool_path, &bin_destination, target_compiler.host);
```

Comment:
It's extremely hard to make sure if we covered all of binaries with `copy_link_binary`. Even if we did it today, I feel like some contributors will use `copy_link` instead of `copy_link_binary` accidentally in the future.

What do you think about adding a `bool` flag to `copy_link` (instead of creating `copy_link_binary`) and handling the debuginfo logic there? It can be harder to miss the logic since they would have to use an incorrect value (`false`) when copying binaries.

Reply:
Seems much better!

Reply:
Thanks for the suggestion! I think this turned out pretty nicely 🙂